### ISSUE TITLE
clang-win: linking errors LNK1120, LNK1112

### ISSUE THREAD

**Issue by hia3** [boostorg/build#156]
I'm trying to build attached example using BB from 1.63.

It builds just fine if `toolset=msvc-14.0`, but with `toolset=clang-win` I'm getting `LNK1120: unresolved external`. -d+2 shows that `user32.lib` is missing in linker `rsp` file.

Building with `address-model=32` gives another linker error.

[clang-min.zip](https://github.com/boostorg/build/files/731788/clang-min.zip)

Thanks.

**Comment by eldiener** [comment]
Which version of clang are you using ?

**Comment by hia3** [comment]
This one - http://llvm.org/pre-releases/win-snapshots/LLVM-4.0.0-r291454-win64.exe

**Comment by eldiener** [comment]
Please give your 'using toolset...', full b2 command, and information about where ( what Boost library ) you are invoking b2 from.

**Comment by hia3** [comment]
My "using" is in `site-config.jam.my` file. Full command in `start_me.bat`. Commands should be invoked in the same directory where `start_me.bat` lives.

**Comment by eldiener** [comment]
The WinMain function is missing starting and ending braces. Please correct and try again.

**Comment by hia3** [comment]
No, they are not missing. This thing called [function-try-block](http://en.cppreference.com/w/cpp/language/function-try-block).

**Comment by eldiener** [comment]
What I am seeing is that when using clang-win, as opposed to msvc, Boost Build is trying to build user32 and gdi32 instead of treating these libs as existing and using them when building test1. I queried about this in the appropriate Boost mailing lists but never received an answer about it from people who are more knowledgeable about the internals of Boost Build.

**Comment by hia3** [comment]
![image](https://cloud.githubusercontent.com/assets/4704208/23179245/404fc17c-f87e-11e6-943f-1bf8c5614833.png)
